modules/channel-web/src/backend/db.ts

```diff
@@ -1,23 +1,109 @@
-import Bluebird from 'bluebird'
 import * as sdk from 'botpress/sdk'
+import { Raw } from 'knex'
 import _ from 'lodash'
 import moment from 'moment'
 import ms from 'ms'
 import uuid from 'uuid'
 
 import { Config } from '../config'
 
+import { DBMessage } from './typings'
+
 export default class WebchatDb {
-  knex: any
+  knex: sdk.KnexExtended
   users: typeof sdk.users
+  private queuedMessages: DBMessage[] = []
+  private queuedConvos: Dic<Raw<any>> = {}
+  private messagePromise: Promise<void | number[]>
+  private convoPromise: Promise<void>
 
   constructor(private bp: typeof sdk) {
     this.users = bp.users
     this.knex = bp['database'] // TODO Fixme
+    setInterval(() => this.flush(), ms('5s'))
+  }
+
+  flush() {
+    this.flushMessages()
+    this.flushConvoUpdates()
+  }
+
+  flushMessages() {
+    if (this.messagePromise || !this.queuedMessages.length) {
+      return
+    }
+
+    const original = this.queuedMessages
+    this.queuedMessages = []
```

Comment:
Please run them in a batch just in case. Don't just process all messages at once.